Source/EasyNetQ/Producer/ClientCommandDispatcherSingleton.cs

```diff
@@ -101,8 +101,8 @@ private void StartDispatcherThread(ConnectionConfiguration configuration)
                 {
                     try
                     {
-                        var channelAction = queue.Take(cancellation.Token);
-                        channelAction();
+                        if (queue.TryTake(out var channelAction, -1, cancellation.Token))
```

Comment:
You can use `Timeout.Infinite` instead of `-1`

Reply:
Fixed as well